Cloudmarque.Azure/Public/IaaS/wvd/New-CmAzWVD.ps1

```
@@ -0,0 +1,110 @@
+	param(
+		[parameter(Mandatory = $true, ParameterSetName = "Settings File")]
+		[String]$SettingsFile,
+		[parameter(Mandatory = $true, ParameterSetName = "Settings Object")]
+		[Object]$SettingsObject
+	)
+
+	$ErrorActionPreference = "stop"
+
+	try {
+
+		if ($PSCmdlet.ShouldProcess((Get-CmAzSubscriptionName), "Deploy Windows Virtual Desktop")) {
+
+			if ($SettingsFile -and -not $SettingsObject) {
+				[array]$SettingsObject = Get-CmAzSettingsFile -Path $SettingsFile
+			}
+			elseif (-not $SettingsFile -and -not $SettingsObject) {
+				Write-Error "No valid input settings." -Category InvalidArgument -CategoryTargetName "SettingsObject"
+            }
+		}
+
+		Write-Verbose "Begining WVD infrastructure object creation."
+		foreach ($Workspace in $SettingsObject.Clone()) {
+			$wvdResourceGroupName = Get-CmAzResourceName -Resource "ResourceGroup" -Architecture "IaaS" -Region $Workspace.workspaceLocation -Name $Workspace.workspaceName
+			Write-Verbose "Generated resource group name: $wvdResourceGroupName"
+			$Workspace.rgName = $wvdResourceGroupName
+		}
+		Write-Verbose "Depoloying WVD infrastructure."
+		New-AzDeployment `
+			-Name "New-CmAzWVD_$(Get-Date -Format 'ddMMyyHHmmss')" `
+			-Location $SettingsObject[0].workspaceLocation `
+			-TemplateFile "C:\Source\cloudmarque-azure-pwsh\Cloudmarque.Azure\Public\IaaS\wvd\New-CmAzWVDHostpools.json" `
+			-settingsObject $SettingsObject
+		Remove-Variable Workspace
+
+		Write-Verbose "Begining WVD host\s object creation."
+		foreach ($Workspace in $SettingsObject.Clone()) {
+			Write-Verbose "Creating hostpool registration key for: $($Workspace.hostpool.hostpoolName)"
+			New-AzWvdRegistrationInfo -ResourceGroupName $Workspace.rgName -HostPoolName $Workspace.hostpool.hostpoolName -ExpirationTime ((Get-Date).AddDays('1'))
+			$Workspace.hostpool.hostpoolToken = (Get-AzWvdRegistrationInfo -ResourceGroupName $Workspace.rgName -HostPoolName $Workspace.hostpool.hostpoolName).Token
+
```

Would be worth performing a check to see if the registration info is returned, if there's no info then we can throw an error describing the issue.

---

Implemented, resolving.